api/services/export-data-2.js

```diff
@@ -1,139 +1,46 @@
-const _ = require('underscore'),
-      objectPath = require('object-path'),
+const { Readable } = require('stream'),
       db = require('../db-pouch'),
       lineage = require('lineage')(Promise, db.medic);
 
-const { Readable } = require('stream'),
-      search = require('search')(Promise, db.medic);
-
 const BATCH = 100;
 
-const SUPPORTED_EXPORTS = ['reports'];
-
-/**
- * Flattens a given object into an object where the keys are dot-notation
- * paths to the flattened values:
- * {
- *   "foo": {
- *     "bar": "smang"
- *   }
- * }
- *
- * becomes:
- *
- * {
- *   "foo.bar": "smang"
- * }
- */
-const flatten = (fields, prepend=[]) => {
-  return Object.keys(fields).reduce((acc, k) => {
-    const path = [...prepend, k];
-
-    if (typeof fields[k] === 'object' && fields[k] && !Array.isArray(fields[k])) {
-      // Recurse into valid objects. We ignore arrays as they are variable
-      // length, so you can't generate a stable header out of them. Instead,
-      // when we convert them into CSV we JSON.stringify them and treat them as
-      // one cell.
-      _.extend(acc, flatten(fields[k], path));
+const MAPPERS = {
+  reports: require('./export/report-mapper'),
+  contacts: require('./export/contact-mapper'),
+  messages: require('./export/message-mapper'),
+};
+
+const joinLine = (csvLine) => csvLine.join(',') + '\n';
+
+const csvLineToString = (csvLine) => {
```

Comment:
I have no strong opinions about this, but interested as to why you pulled this out of the class it was in. Did something start breaking? Or you just don't really want any code that is vaguely OOPey in JS? (I also am not an OOPey person, but I figured since I had to write a class here I figured in for a penny...)

Reply:
Yeah it was just a code style thing... now that this module has very little in it it doesn't change the scope much to hoist the functions out and it's just more JSey to have first class functions rather than static members.